pkg/test/step.go

```diff
@@ -122,8 +122,8 @@ func (s *Step) CheckResource(expected runtime.Object, namespace string) []error
 
 		err = s.Client.List(context.TODO(), actual, listOptions...)
 
-		for _, item := range actual.Items {
-			actuals = append(actuals, &item)
+		for index := range actual.Items {
```

Comment:
maybe it's too early but I read the description several times and I still feel like these two implementations should do the same thing 🙈

Reply:
They aren’t the same, see this post for some background: https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables

Before I took the address of the loop variable which is modified for each iteration of the loop, `actuals` was a slice of the same pointer over and over.

Now, I take the address of the actual item from the list using the index, so the loop variable isn’t used.